docs/ansible_turbo_mode.rst

```diff
@@ -0,0 +1,200 @@
+
+.. code-block:: python
+
+  from ansible_module.turbo.module import AnsibleTurboModule as AnsibleModule
+
+You can also use the ``functools.lru_cache()`` decorator to ask Python to cache
+the result of an operation, like a network session creation.
+
+Finally, if some of the dependeded libraries are large, it may be nice
+to defer your module imports, and do the loading AFTER the
+``AnsibleTurboModule`` instance creation.
+
+.. code-block:: python
+
+  from ansible_module.turbo.module import AnsibleTurboModule as AnsibleModule
+
+  class MyModule(AnsibleModule):
+      ...
+
+  def main():
+    instance = MyModule()
+    import big_library
+
+
+.. note:: AnsibleTurboModule depends on Python 3.6 or greater
+
+
+Ansible Turbo with Kubernetes.core
+==================================
+
+Ansible Turbo is already implemented in ``kubernetes.core`` from version 2.0.0. You need to just
+install ``cloud.common`` collection and ``aiohttp`` as per documentation.
+
+Example
+=======
+
+The Ansible module is slightly different while using AnsibleTurboModule.
+Here are examples with Kubernetes, OpenStack, and VMware.
+
+These examples use ``functools.lru_cache`` that is the Python core since 3.3.
+``lru_cache()`` decorator will managed the cache. It uses the function parameters
+as unicity criteria.
+
+- Integration with Kubernetes collection: https://github.com/ansible-collections/kubernetes.core/pull/68
+- Integration with OpenStack Collection: https://github.com/goneri/ansible-collections-openstack/commit/53ce9860bb84eeab49a46f7a30e3c9588d53e367
+- Integration with VMware Collection: https://github.com/goneri/vmware/commit/d1c02b93cbf899fde3a4665e6bcb4d7531f683a3
+
+The background service
+======================
+
+The daemon kills itself after 15s, and communication are done
+through an Unix socket.
+It runs in one single process and uses ``asyncio`` internally.
+Consequently you can use the ``sync`` keyword in your Ansible module.
+This will be handy if you interact with a lot of remote systems
+at the same time.
+
+Security impact
+===============
+
+``ansible_module.turbo`` open an Unix socket to interact with the background service.
+We use this service to open the connection toward the different target systems.
+
+This is similar to what SSH does with the sockets.
+
+Keep in mind that:
+
+- All the modules can access the same cache. Soon an isolation will be done at the collection level (https://github.com/ansible-collections/cloud.common/pull/17)
+- A task can loaded a different version of a library and impact the next tasks.
+- If the same user runs two ``ansible-playbook`` at the same time, they will have access to the same cache.
+
+When a module stores a session in a cache, it's a good idea to use a hash of the authentication information to identify the session.
+
+.. note:: You may want to isolate your Ansible environemt in a container, in this case you can consider https://github.com/ansible/ansible-builder
```

Comment:
It's possible I may not be fully understanding execution environments, but my sense was that modules are not being run in a containerized environment on the managed node. If that's the case then neither the turbo mode daemon nor the module interaction with the daemon is occurring within a container, right? This note here would seem a little misleading assuming I'm understanding things correctly.